src/test/test_job.ts

```
@@ -61,6 +61,84 @@ describe('Job', function() {
     });
   });
 
+  it('should move parent job to waiting-children', async () => {
+    const values = [
+      { idx: 0, bar: 'something' },
+      { idx: 1, baz: 'something' },
+      { idx: 2, qux: 'something' },
+    ];
+
+    const parentQueueName = 'parent-queue';
+
+    let childrenProcessor,
+      parentProcessor,
+      processedChildren = 0;
+    const processingChildren = new Promise<void>((resolve, reject) => [
+      (childrenProcessor = async (job: Job) => {
+        processedChildren++;
+        if (processedChildren === values.length) {
+          resolve();
+        }
+      }),
+    ]);
+
+    const processingParent = new Promise<void>((resolve, reject) => [
+      (parentProcessor = async (job: Job) => {
+        try {
+          const movedToWaitingChildren = await job.moveToWaitingChildren('0');
```

I am still a bit confused that we can allow this call from inside the worker processor, when the call ends it will try to move the job to "completed" from "active", but since we manually moved the job to "waiting-children" it should raise an error somewhere.

---

sure, in moveToFinished I'm also checking if there are pending dependencies, in case there are, throws an error

---

yes I saw that, but the current test code, does it not trigger that  error? since the job has been moved and when the processor call returns the worker will try to move the job to completed and your check should return an error. So two things here to think about, how to handle this error (I suspect it is erroring silently now), and two, what is the proper way to do this without raising an error. Somehow we have to tell the worker that we have moved the job to waiting-children so that it does not matter anymore if the job was in active, and in this case do a no op. Or maybe "moveToWaitingChildren" should be a method that can only be used when processing jobs manually. Do you have a need for this use case where it has to be called inside the processor function?